src/jllvm/object/ClassObject.hpp

```diff
@@ -45,6 +45,14 @@ enum class Visibility : std::uint8_t
     Protected = 0b11
 };
 
+/// Initialization status of a class
+enum class InitializationStatus : std::uint8_t
+{
+    Uninitialized,
+    UnderInitialization,
+    Initialized,
+};
```

Comment:
Can we assign values to these values purely to signal that their exact values are significant? There is LLVM IR that reads this enum and relies on `Uninitialized` being 0. I think you could also move the enum closer to `ClassObject` in the file